sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala

```diff
@@ -1085,6 +1085,122 @@ class SparkE2EQueryITest
+                     |   }
+                     | }
+                     |""".stripMargin
+
+    val objectNode2 = objectMapper.readValue(rawItem2, classOf[ObjectNode])
+
+    container.createItem(objectNode2).block()
+
+    val cfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+      "spark.cosmos.accountKey" -> cosmosMasterKey,
+      "spark.cosmos.database" -> cosmosDatabase,
+      "spark.cosmos.container" -> cosmosContainer,
+      "spark.cosmos.read.partitioning.strategy" -> "Restrictive",
+      "spark.cosmos.read.customQuery" -> "SELECT VALUE c.nestedObject.prop1 FROM c"
+    )
+
+    val df = spark.read.format("cosmos.oltp").options(cfg).load()
+    val rowsArray = df.collect()
+    rowsArray should have size 2
+
+    val item = rowsArray(0)
+    item.getAs[Int]("_value") shouldEqual 5
+
+    val item2 = rowsArray(1)
+    item2.getAs[Int]("_value") shouldEqual 5
+  }
+
+  "spark query" can "execute query with VALUE function ad ORDER BY" in {
+    val cosmosEndpoint = TestConfigurations.HOST
+    val cosmosMasterKey = TestConfigurations.MASTER_KEY
+
+    val id = UUID.randomUUID().toString
+
+    val rawItem = s"""
+                     | {
+                     |   "id" : "${id}",
+                     |   "nestedObject" : {
+                     |     "prop1" : 5,
+                     |     "prop2" : "6"
+                     |   }
+                     | }
+                     |""".stripMargin
+
+    val objectNode = objectMapper.readValue(rawItem, classOf[ObjectNode])
+
+    val container = cosmosClient.getDatabase(cosmosDatabase).getContainer(cosmosContainer)
+    container.createItem(objectNode).block()
+
+    val id2 = UUID.randomUUID().toString
+
+    val rawItem2 = s"""
+                      | {
+                      |   "id" : "${id2}",
+                      |   "nestedObject" : {
+                      |     "prop1" : 8,
+                      |     "prop2" : "7"
+                      |   }
+                      | }
+                      |""".stripMargin
+
+    val objectNode2 = objectMapper.readValue(rawItem2, classOf[ObjectNode])
+
+    container.createItem(objectNode2).block()
+
+    val cfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+      "spark.cosmos.accountKey" -> cosmosMasterKey,
+      "spark.cosmos.database" -> cosmosDatabase,
+      "spark.cosmos.container" -> cosmosContainer,
+      "spark.cosmos.read.partitioning.strategy" -> "Restrictive",
+      "spark.cosmos.read.customQuery" -> "SELECT VALUE c.nestedObject.prop1 FROM c ORDER BY c.nestedObject.prop2"
+    )
+
+    val df = spark.read.format("cosmos.oltp").options(cfg).load()
+
+    // sorting on the Spark level is necessary because each custom query is only ever sent against an EPK
+    // smaller or equal to a physical partition - so for each partition results would be sorted but
+    // re-ordering on the spark level is needed to get ordering across partitions
+    val rowsArray = df.orderBy("_value").collect()
```

Comment:
We can't - when using custom queries they are sent as is to the backend. 

Will change the comment - but what I tried to explain is that I added the spark-level order bv to have an easy way to compare the content

both row could come in any order - because the order by prop 2 would only result in predictive order if all documents are on the same physical partition. By adding the spark-level order by I can use the simple content validation assuming a certain global order. Does that make sense?

Reply:
Make sense

But then my question is in real life, if customer is using orderby query, then does it mean in spark, we actually can not really guarantee the global order?